src/lib.rs

```diff
@@ -125,6 +123,25 @@ fn bits_in_branch_sanity_checks() {
                                        0b0100));
 }
 
+fn layout(size: usize, align: usize) -> Layout {
+    Layout::from_size_align(size, align).expect("Invalid memory layout.")
```

Comment:
I feel like these `expect`s change the semantics of this a bit.  Still might accept them but I should look over the new allocation API.

Reply:
My understanding was that the possibility of an invalid `Layout` seems remote, and I guess there is nothing else to do but go down in flames if it happens. It'd probably be an application error... We can fix the error messages in alter PR if you find something better to say. The `expect` should also add a bunch of context in case of failure.

Reply:
After looking at this, I think several parts of the code should use `Layout::for_value` instead of `from_size_align`.  I know that will mean a slightly more invasive change, but it will also mean less meaningless error handling.  In the cases where we need to use `from_size_align`, maybe it will be clearer why it could fail.  I agree it's good that we'll get a bunch of context.

If it's possible to do this without wrapping these functions, that would be best.

Reply:
After three failed attempts at using `Layout::for_value`, I believe this is best left as is. I'll let you try it and find a recipe that suits your taste.